crates/forge/src/compatibility_check.rs

```diff
@@ -249,4 +321,116 @@ mod tests {
+                cache_dir: Utf8PathBuf::default(),
+                contracts_data: ContractsData::try_from(contracts).unwrap(),
+                environment_variables: HashMap::default(),
+            }),
+            output_config: Arc::new(OutputConfig {
+                detailed_resources: false,
+                execution_data_to_save: ExecutionDataToSave::default(),
+            }),
+        };
+        assert!(check_sierra_gas_version_requirement(&forge_config).is_ok());
+    }
+
+    #[test]
+    fn sierra_gas_version_requirement_no_contracts() {
+        let forge_config = ForgeConfig {
+            test_runner_config: Arc::new(TestRunnerConfig {
+                exit_first: false,
+                fuzzer_runs: NonZeroU32::new(256).unwrap(),
+                fuzzer_seed: 12345,
+                max_n_steps: None,
+                is_vm_trace_needed: false,
+                tracked_resource: ForgeTrackedResource::SierraGas,
+                cache_dir: Utf8PathBuf::default(),
+                contracts_data: ContractsData::default(),
+                environment_variables: HashMap::default(),
+            }),
+            output_config: Arc::new(OutputConfig {
+                detailed_resources: false,
+                execution_data_to_save: ExecutionDataToSave::default(),
+            }),
+        };
+        assert!(check_sierra_gas_version_requirement(&forge_config).is_ok());
+    }
+
+    #[test]
+    fn sierra_gas_version_requirement_fail() {
+        let mut contracts = HashMap::new();
+        contracts.insert(ContractName::from("MockedContract"),
+                         (StarknetContractArtifacts {
+                             sierra: String::from("{\"sierra_program\":[\"0x1\",\"0x6\",\"0x0\"],\"sierra_program_debug_info\":{\"type_names\":[],\"libfunc_names\":[],\"user_func_names\":[]},\"contract_class_version\":\"0.1.0\",\"entry_points_by_type\":{\"EXTERNAL\":[],\"L1_HANDLER\":[],\"CONSTRUCTOR\":[]},\"abi\":[]}"),
+                             casm: String::new()},
+                          Utf8PathBuf::from("whatever")
+                         ));
+
+        let forge_config = ForgeConfig {
+            test_runner_config: Arc::new(TestRunnerConfig {
+                exit_first: false,
+                fuzzer_runs: NonZeroU32::new(256).unwrap(),
+                fuzzer_seed: 12345,
+                max_n_steps: None,
+                is_vm_trace_needed: false,
+                tracked_resource: ForgeTrackedResource::SierraGas,
+                cache_dir: Utf8PathBuf::default(),
+                contracts_data: ContractsData::try_from(contracts).unwrap(),
+                environment_variables: HashMap::default(),
+            }),
+            output_config: Arc::new(OutputConfig {
+                detailed_resources: false,
+                execution_data_to_save: ExecutionDataToSave::default(),
+            }),
+        };
+        assert!(check_sierra_gas_version_requirement(&forge_config).is_err());
```

Comment:
Can we do `unwrap_err` and assert the error contents?

Reply:
Added, please note that there is also a e2e test that asserts this `sierra_gas_with_older_scarb`